packages/core/src/registries/designTokenRegistry.ts

```diff
@@ -13,12 +13,22 @@ export const defineDesignTokens = (designTokenDefinition: DesignTokensDefinition
 
 const templateStringRegex = /\${(.+?)}/g;
 
-export const getDesignTokenRegistration = (breakpointValue: string) => {
+export const getDesignTokenRegistration = (breakpointValue: string, variableName: string) => {
   if (!breakpointValue) return breakpointValue;
 
   let resolvedValue = '';
   for (const part of breakpointValue.split(' ')) {
-    const tokenValue = templateStringRegex.test(part) ? resolveSimpleDesignToken(part) : part;
+    let tokenValue = '';
+    if (variableName === 'cfBorder' && templateStringRegex.test(part)) {
+      tokenValue = resolveBorderDesignToken(part);
+      const { width, color, style } =
+        (tokenValue as unknown as DesignTokensDefinition['borders']) || {
+          width: '1px',
+          style: 'outside',
+          color: 'rgba(0,0,0,1)',
+        };
+      tokenValue = `${width} ${style} ${color}`;
+    } else tokenValue = templateStringRegex.test(part) ? resolveSimpleDesignToken(part) : part;
```

Comment:
I think the type casting can be avoided here.  It seems to stem from the return type being wrong for `resolveBorderDesignToken` (string instead of the border object)

Here's a possible way to rewrite this without any type casting (although I didn't test this)

```suggestion
    if (variableName === 'cfBorder' && templateStringRegex.test(part)) {
      const borderTokenValue = resolveBorderDesignToken(part);
      const { width, color, style } = borderTokenValue || {
        width: '1px',
        style: 'outside',
        color: 'rgba(0,0,0,1)',
      };
      resolvedValue += `${width} ${style} ${color} `;
      continue;
    }
    const tokenValue = templateStringRegex.test(part) ? resolveSimpleDesignToken(part) : part;
    resolvedValue += `${tokenValue} `;
```